CMakeLists.txt

```diff
@@ -1,62 +1,44 @@
+cmake_minimum_required(VERSION 3.5)
 project(visualization.shadertoy)
 
-cmake_minimum_required(VERSION 2.6)
-
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR})
 
 find_package(Kodi REQUIRED)
 find_package(p8-platform REQUIRED)
 
 add_subdirectory(lib/kissfft)
 
-set(DEPLIBS kissfft)
-
 if (NOT WIN32)
-  if(APPLE)
-    set(DEPLIBS ${DEPLIBS} "-framework CoreVideo")
-  endif()
-
-  add_options(CXX ALL_BUILDS -std=c++11)
-
   if(APP_RENDER_SYSTEM STREQUAL "gl" OR NOT APP_RENDER_SYSTEM)
-    find_package(OpenGL)
-    if(OPENGL_FOUND)
-      set(INCLUDES ${OpenGL_INCLUDE_DIR})
-      add_definitions(-DHAS_OPENGL)
-      set(DEPLIBS ${DEPLIBS} ${OPENGL_LIBRARIES})
-      set(APP_RENDER_SYSTEM gl)
-    endif()
+    find_package(OpenGl REQUIRED)
+    set(DEPLIBS ${OPENGL_LIBRARIES})
+    set(includes ${OPENGL_INCLUDE_DIR})
+    add_definitions(${OPENGL_DEFINITIONS})
+  elseif(APP_RENDER_SYSTEM STREQUAL "gles")
+    find_package(OpenGLES REQUIRED)
+    set(DEPLIBS ${OPENGLES_LIBRARIES})
+    set(includes ${OPENGLES_INCLUDE_DIR})
+    add_definitions(${OPENGLES_DEFINITIONS})
   endif()
 
-  if(APP_RENDER_SYSTEM STREQUAL "gles" OR NOT APP_RENDER_SYSTEM)
-    find_package(OpenGLES2)
-    if(OPENGLES2_FOUND)
-      set(INCLUDES ${OPENGLES2_INCLUDE_DIR})
-      add_definitions(-DHAS_GLES2 -DHAS_GLES ${OpenGLES2_CFLAGS_OTHER})
-      set(DEPLIBS ${DEPLIBS} ${OPENGLES2_LIBRARIES})
-      set(APP_RENDER_SYSTEM gles)
+  include_directories(${GLM_INCLUDE_DIR}
+                      ${KODI_INCLUDE_DIR}/.. # Hack way with "/..", need bigger Kodi cmake rework to match right include ways
+                      ${PROJECT_SOURCE_DIR}/lib)
+
+  if(NOT ${CORE_SYSTEM_NAME} STREQUAL "")
+    if(${CORE_SYSTEM_NAME} STREQUAL "osx" OR
+       ${CORE_SYSTEM_NAME} STREQUAL "ios" OR
+       ${CORE_SYSTEM_NAME} STREQUAL "darwin_embedded")
+      list(APPEND DEPLIBS "-framework CoreVideo")
     endif()
   endif()
-  if(NOT APP_RENDER_SYSTEM)
-    message(FATAL_ERROR "No rendering API found. Bailing")
```

Comment:
Should be in the else branch of the above if `APP_RENDER_SYSTEM`

Reply:
This is not required, thats why removed.
cmake's find_package generates following if not present:
```
CMake Error at /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenGl (missing: OPENGL_gl_LIBRARY OPENGL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  FindOpenGl.cmake:31 (find_package_handle_standard_args)
  CMakeLists.txt:13 (find_package)
```

Reply:
There is still a case left, `APP_RENDER_SYSTEM` not empty, not gl and not gles.
Either add an else branch or replace `elseif(APP_RENDER_SYSTEM STREQUAL "gles")` with `else()`.

Reply:
I change to `else()`.

As question, which render system except "GL" and "GLES" could still be defined?
If anything else came along, there would be something bigger about Kodi and the build system broken.